From b7b380ce51650f19e10ceff1b70a92089b6738ba Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 12 Oct 2017 01:02:55 +0200
Subject: [PATCH] For elf output, our final executable format does not have
 .init/.fini sections, so some things from elfos.h must be undone to make
 global constructors/destructors work.

---
 gcc/config/m68k/mint.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/gcc/config/m68k/mint.h b/gcc/config/m68k/mint.h
index 6cf6c868700..61385f29c83 100644
--- a/gcc/config/m68k/mint.h
+++ b/gcc/config/m68k/mint.h
@@ -174,6 +174,17 @@ along with GCC; see the file COPYING3.  If not see
 #define JUMP_TABLES_IN_TEXT_SECTION 1
 
 #define EH_TABLES_CAN_BE_READ_ONLY 1
+
+/*
+ * our object format is elf, but not our executable format
+ */
+#undef HAVE_INITFINI_ARRAY_SUPPORT
+#define HAVE_INITFINI_ARRAY_SUPPORT 0
+
+#undef OBJECT_FORMAT_ELF 
+#undef INIT_SECTION_ASM_OP
+#undef FINI_SECTION_ASM_OP
+
 #else
 #define BSS_SECTION_ASM_OP "\t.bss"
 
-- 
2.24.0

